### PR TITLE
Several e2e fixes

### PR DESCRIPTION
**Pull request by Luap99**
Split out some fixes from #18163 that can be merged without ginkgo v2.
this should make review easier.  Please see the individual commits for details.

#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

File: test/e2e/run_volume_test.go
Comment (edsantiago) [inline test/e2e/run_volume_test.go:681]: super-duper-minor nit: instead of hardcoding, might it be possible to use `Caller()`? Again, super-minor, it seems impossible for this to fail: even if the source file is renamed, the error would be caught immediately. It's just one of my neuroses.

**Comment by edsantiago** [comment]
Lots of really nice catches. Thanks for breaking them out into a separate PR with separate commits.

**Review by vrothberg — APPROVED** [review]
LGTM

File: test/e2e/run_volume_test.go
Comment (Luap99) [reply test/e2e/run_volume_test.go:681]: Mhh searching around there is `CurrentGinkgoTestDescription().FileName`, I wasn't ware of such functionality before.
If I have to repush I can add it.

**Comment by edsantiago** [comment]
Sigh.... needs to wait for #18282 which is currently stuck in CI-flake land.

**Comment by Luap99** [comment]
Test are failing because of #18282, I will rebase once this merges and apply the nit from @edsantiago

**Comment by edsantiago** [comment]
LGTM and tests are green. @containers/podman-maintainers PTAL.

**Comment by mheon** [comment]
LGTM. I'm just going to press the button in case the bot has not recovered yet.

**Pull request merged**
